pkg/pillar/types/zedroutertypes.go

```diff
@@ -1897,3 +1897,8 @@ type VifIPTrig struct {
 func (vifIP VifIPTrig) Key() string {
 	return vifIP.MacAddr
 }
+
+// ClientUUID - UUID advertised by client process
+type ClientUUID struct {
```

Comment:
Very good but I'd suggest naming this struct something more general like OnboardingStatus.
In addition to the DeviceUUID we might want to feed the device certificate through this pubsub; I'm thinking of how we'd get to the point when device-steps.sh goes away and client.go does all of:
 - check if we have a device certificate; if so proceed with selfRegister/getUuuid as now
 - if not, then check if we have current time from NTP (requires having a network etc); can potentially do that by trying an TLS connection since it the time is off by a year the TLS connection will fail due to the cert NotBefore/NotAfter constraints. Once it knows it has reasonable time the client.go code can generate the devie certificate.

In that model all the agents start, and all the ones which run after client today in device-steps.sh would just wait for the ClientUUID/OnboardStatus to be published.

Maybe we only need the DeviceUUID in that case as well, but a more generic name for the struct can't hurt.
FWIW the only thing I don't know for sure how to do in that picture is to generate the TPM-based key pair/cert using the go code; there are examples of how to generate key pairs/certs using the go packages,